backend/backtest_service.py

```
@@ -100,8 +103,11 @@ def backtest_strategy(strategy_id: int, symbol: str = 'AAPL', days: int = 30, in
 
 def list_backtest_results(limit: int = 10) -> List[Dict[str, Any]]:
     """List recent backtest results from KB"""
+    if not KB_AVAILABLE or not kb_service:
+        return []
     try:
         results = kb_service.search("backtest", limit=limit)
         return results
     except Exception as e:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)